### PR TITLE
added metrics for block upload stats

### PR DESCRIPTION
**Pull request by aldernero**
This PR adds some basic counters around completed block uploads. We can see failed upload information in logs, but currently successful block uploads don't produce any logs unless using the debug log level.

The idea is to be able to track
1. blocks successfully uploaded (count) - `cortex_block_upload_api_blocks_total`
2. total bytes of uploaded blocks - `cortex_block_upload_api_bytes_total`
3. total number of files in uploaded blocks - `cortex_block_upload_api_files_total`

These can be helpful when correlating changes seen in the compactor or object storage.

#### Checklist

- [X] Tests updated
- [N/A] Documentation added
- [X] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: pkg/compactor/compactor.go [hunk 1]
```diff
@@ -367,6 +371,18 @@ func newMultitenantCompactor(
 			Help:        blocksMarkedForDeletionHelp,
 			ConstLabels: prometheus.Labels{"reason": "compaction"},
 		}),
+		blockUploadBlocks: promauto.With(registerer).NewCounter(prometheus.CounterOpts{
+			Name: "cortex_block_upload_blocks_total",
+			Help: "Total number of blocks successfully uploaded and validated using the block upload API.",
+		}),
+		blockUploadBytes: promauto.With(registerer).NewCounter(prometheus.CounterOpts{
+			Name: "cortex_block_upload_bytes_total",
+			Help: "Total number of bytes from successfully uploaded blocks using block upload API.",
```
Comment (lamida) [inline pkg/compactor/compactor.go:380]:
```suggestion
			Help: "Total number of bytes from successfully uploaded and validated blocks using block upload API.",
```

File: pkg/compactor/compactor.go [hunk 2]
```diff
@@ -367,6 +371,18 @@ func newMultitenantCompactor(
...
+		}),
+		blockUploadFiles: promauto.With(registerer).NewCounter(prometheus.CounterOpts{
+			Name: "cortex_block_upload_files_total",
+			Help: "Total number of files from successfully uploaded blocks using block upload API.",
```
Comment (lamida) [inline pkg/compactor/compactor.go:384]:
```suggestion
			Help: "Total number of files from successfully uploaded and validated blocks using block upload API.",
```

**Review by charleskorn — APPROVED** [review]
LGTM, modulo Jon's suggestions

**Comment by bboreham** [comment]
There is currently `cortex_ingester_shipper_uploads_total`; I can see me getting confused that this new one is compactor-only.
`cortex_block_upload_blocks_total` would work for both, but we can't make a breaking change to the existing one.

**Comment by bboreham** [comment]
You might want these to be per-tenant.

**Review by pracucci — APPROVED** [review]
LGTM (modulo a comment to metric name)

File: CHANGELOG.md [hunk 3]
```diff
@@ -72,6 +72,7 @@
 * [ENHANCEMENT] Compactor: Delete bucket-index, markers and debug files if there are no blocks left in the bucket index. This cleanup must be enabled by using `-compactor.no-blocks-file-cleanup-enabled` option. #5648
 * [ENHANCEMENT] Ingester: reduce memory usage of active series tracker. #5665
 * [ENHANCEMENT] Store-gateway: added `-store-gateway.sharding-ring.auto-forget-enabled` configuration parameter to control whether store-gateway auto-forget feature should be enabled or disabled (enabled by default). #5702
+* [ENHANCEMENT] Compactor: added block upload counters `cortex_block_upload_blocks_total`, `cortex_block_upload_bytes_total`, and `cortex_block_upload_files_total`. #5738
```
Comment (treid314) [inline CHANGELOG.md:75]: nit: Do you want to mention these are per-tenant?

Comment (aldernero) [reply CHANGELOG.md]: thanks for catching that, I also updated the counter names

**Pull request merged**